src/row.js

```diff
@@ -553,14 +451,173 @@ Row.prototype.delete = function(callback) {
  *   var apiResponse = data[0];
  * });
  */
-Row.prototype.deleteCells = function(columns, callback) {
+Row.prototype.deleteCells = function(columns, gaxOptions, callback) {
+  if (is.fn(gaxOptions)) {
+    callback = gaxOptions;
+    gaxOptions = {};
+  }
+
   var mutation = {
     key: this.id,
     data: arrify(columns),
     method: Mutation.methods.DELETE,
   };
 
-  this.parent.mutate(mutation, callback);
+  this.table.mutate(mutation, gaxOptions, callback);
+};
+
+/**
+ * Check if the table row exists.
+ *
+ * @param {object} [gaxOptions] Request configuration options, outlined here:
+ *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ * @param {function} callback The callback function.
+ * @param {?error} callback.err An error returned while making this
+ *     request.
+ * @param {boolean} callback.exists Whether the row exists or not.
+ *
+ * @example
+ * row.exists(function(err, exists) {});
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * row.exists().then(function(data) {
+ *   var exists = data[0];
+ * });
+ */
+Row.prototype.exists = function(gaxOptions, callback) {
+  if (is.fn(gaxOptions)) {
+    callback = gaxOptions;
+    gaxOptions = {};
+  }
+
+  this.getMetadata(gaxOptions, function(err) {
+    if (err) {
+      if (err instanceof RowError) {
```

Comment:
I'm curious as to why this is sometimes `error.name === 'ThingError'` sometimes `err.code === 5`, and sometimes `err instanceof RowError`.

Reply:
I found one `err.name === 'FamilyError'` check, and I changed that to `err instanceof FamilyError`. Were there others you found?

For this line we're commenting on (`err instanceof RowError`), when a row doesn't exist, that's the error our API returns from `row.get()`, which is called by `this.getMetadata()`.

For the other places, the code 5 is what GAPIC returns, a raw GRPC error object.

Reply:
I haven't found any others. Makes sense